pandas/io/parsers.py

```diff
@@ -2875,20 +2877,34 @@ def __init__(self, f, colspecs, delimiter, comment):
                 raise TypeError('Each column specification must be '
                                 '2 element tuple or list of integers')
 
-    def get_rows(self, n):
-        rows = []
-        for i, row in enumerate(self.f, 1):
-            rows.append(row)
-            if i >= n:
+    def get_rows(self, n, skiprows=None):
+        """
+        We distinguish buffer_rows (the first <= n lines)
+        from the rows returned to detect_colspecs because
+        it's simpler to leave the other locations with
+        skiprows logic alone than to modify them to deal
+        with the fact we skipped some rows here as well.
+        """
```

Comment:
Ah.  This wasn't really meant to be a docstring, just a multiline comment.. but I can add a docstring, I guess. :-)